### ISSUE TITLE
e2e: cni? create two containers with the same IP

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#18399]
Resurrection of #7096, but as new issue to establish clean context. This is only happening with **cni**, not netavark. And only on **create**, never on run. So far.
```
  Podman create two containers with the same IP
...
# podman [options] create --log-driver k8s-file --name test1 --ip 10.88.30.100 quay.io/libpod/alpine:latest sleep 999
406b928fe0375988d1f219f7a3567385dd270817d633ae4b919c2631ca7a1dd7
# podman [options] create --log-driver k8s-file --name test2 --ip 10.88.30.100 quay.io/libpod/alpine:latest ip addr
c76d84306fb9947762a5bd7ae84bbffda08f3e737a45e6d7915a5b2e7a0782d0
# podman [options] start test1
test1
# podman [options] inspect --format {{.State.Status}} {{.NetworkSettings.IPAddress}} test1
running 10.88.30.100
^^^^^^^^^^^^^^^^^^^^ So far, so good

# podman [options] start -a test2
...
2: eth0@if1038: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1500 qdisc noqueue state UP 
    link/ether 82:d7:1c:fb:bd:bd brd ff:ff:ff:ff:ff:ff
    inet 10.88.30.100/16 brd 10.88.255.255 scope global eth0
         ^^^^^^^^^^^^^^^ NOT GOOD!  Also, WHY /16??? Is this a clue?
...
```
Test fails because second container should refuse to start.

* fedora-36 : int podman fedora-36 root container sqlite
    * [03-28-2023 22:06](https://api.cirrus-ci.com/v1/artifact/task/6579579771420672/html/int-podman-fedora-36-root-container-sqlite.log.html#t--Podman-create-two-containers-with-the-same-IP--1) in Podman create with --ip flag [It] Podman create two containers with the same IP
    * [03-28-2023 11:11](https://api.cirrus-ci.com/v1/artifact/task/5157108782465024/html/int-podman-fedora-36-root-container-sqlite.log.html#t--Podman-create-two-containers-with-the-same-IP--1) in Podman create with --ip flag [It] Podman create two containers with the same IP
* fedora-36 : int remote fedora-36 root host sqlite [remote]
    * [04-24 12:44](https://api.cirrus-ci.com/v1/artifact/task/6286884284923904/html/int-remote-fedora-36-root-host-sqlite.log.html#t--Podman-create-two-containers-with-the-same-IP--1) in Podman create with --ip flag [It] Podman create two containers with the same IP
* fedora-37 : int podman fedora-37 root container sqlite
    * [04-27 13:36](https://api.cirrus-ci.com/v1/artifact/task/4968190518755328/html/int-podman-fedora-37-root-container-sqlite.log.html#t--Podman-create-two-containers-with-the-same-IP--1) in Podman create with --ip flag [It] Podman create two containers with the same IP

**Comment by Luap99** [comment]
Wait I thought f37 means netavark, so I was fooled by that.

Ok yeah quick look in the e2e tests show that we do something that this is completely wrong??? https://github.com/containers/podman/pull/1878/commits/55508c11856c612186d685a4e217c462f3ca2416
This is not safe in parallel testing and can cause all sorts of weird flakes. If another test deletes this directory when the between the first and second container start then it will work and cause the flake.

**Comment by edsantiago** [comment]
"f37 means netavark" -- up until last week, #18326. Thanks for the quick fix!
